src/keywords/optionaldouble.cpp

```diff
@@ -55,21 +57,30 @@ std::string OptionalDoubleKeyword::textWhenNull() const { return textWhenNull_;
 // Deserialise from supplied LineParser, starting at given argument offset
 bool OptionalDoubleKeyword::deserialise(LineParser &parser, int startArg, const CoreData &coreData)
 {
-    if (parser.hasArg(startArg))
+    try
     {
+        if (!parser.hasArg(startArg))
+            return false;
+
         auto x = parser.argd(startArg);
         if (!setData(x))
         {
             if (maximumLimit_)
-                Messenger::error("Value {} is out of range for keyword. Valid range is {} <= n.\n", x, maximumLimit_.value());
+                Messenger::error("Value {} is out of range for keyword '{}'. Valid range is {} <= n.\n", x, name(),
+                                 maximumLimit_.value());
 
             return false;
```

Comment:
The `Messenger::error()` function returns `false` so we can do this:
```suggestion
                return Messenger::error("Value {} is out of range for keyword '{}'. Valid range is {} <= n.\n", x, name(),
                                        maximumLimit_.value());
```